File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        // https://www.cs.utexas.edu/~moore/best-ideas/mjrty/
        int max = nums[0];
        int count = 1;
        for (int i = 1; i < nums.length; i++) {
            if (count == 0) {
                max = nums[i];
                count++;
            } else if (nums[i] == max) {
                count++;
            } else {
                count--;
            }
        }
        
        return max;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # https://www.cs.utexas.edu/~moore/best-ideas/mjrty/
        maximum = None
        count = 0
        for num in nums:
            if count == 0:
                maximum = num
                count += 1
            elif num == maximum:
                count += 1
            else:
                count -= 1
        
        return maximum